script/job_generator/versions.py

```diff
@@ -30,11 +29,40 @@
     ],
     HIPCC: ["5.0", "5.1", "5.2", "5.3", "5.4", "5.5"],
     ICPX: ["2023.1.0", "2023.2.0"],
+    # Defines all backends, which will be enabled.
+    # Special cases for ALPAKA_ACC_GPU_CUDA_ENABLE and ALPAKA_ACC_GPU_HIP_ENABLE.
+    # The will be combine with nvcc and hip versions.
     BACKENDS: [
-        #         ALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLE,
-        ALPAKA_ACC_GPU_CUDA_ENABLE,
-        ALPAKA_ACC_GPU_HIP_ENABLE,
-        ALPAKA_ACC_SYCL_ENABLE,
+        # ALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLE,
+        # nvcc
+        [
+            ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLE,
+            ALPAKA_ACC_CPU_B_SEQ_T_THREADS_ENABLE,
+            ALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLE,
+            ALPAKA_ACC_CPU_B_SEQ_T_OMP2_ENABLE,
+            ALPAKA_ACC_GPU_CUDA_ENABLE,
+        ],
+        # clang-cuda
+        # OpenMP is not supported for clang as cuda compiler
+        # https://github.com/alpaka-group/alpaka/issues/639
+        # therefore a extra combination is required
+        [
+            ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLE,
```

Comment:
How can I see that this is for clang-cuda except by the comment? So I am wondering how the generator know that for clang-cuda this configuration should be used and not the first entry in `BACKENDS`?

Reply:
You cannot see it directly in code, because it is decided by filter rules. So `clang-cuda` requires `ALPAKA_ACC_GPU_CUDA_ENABLE` and does not allow the `OMP2` backends:

https://github.com/alpaka-group/alpaka/blob/9a30ae807d761bc0bf278735714a5ac10a04ae93/script/job_generator/alpaka_filter.py#L48-L56

You can check if a compiler backend passes the filter with `ajc-validate` tool (except the alpaka filter, I need to implement support for this :thinking: ).